jdaviz/configs/specviz/tests/test_helper.py

```diff
@@ -248,7 +254,7 @@ def test_load_spectrum_list_directory(tmpdir, specviz_helper):
 
     with pytest.warns(UserWarning, match='SRCTYPE is missing or UNKNOWN in JWST x1d loader'):
         specviz_helper.load_spectrum(data_path)
-    assert len(specviz_helper.app.data_collection) == 3
+    assert len(specviz_helper.app.data_collection) == 41
```

Comment:
is this actually expected to be 41?  This same test has also changed on main to give a length of 40 (failing when expecting 3)... 🤔

Reply:
Did someone dump a bunch of new files into the ZIP file we download for the test? 😱

Reply:
Thanks for the summary @kecnry. The new behavior after this PR is implemented is to load all of the available extensions within the directory (39 total), and make concatenated spectra from each FITS file (2 additional), making 41. Though I'm not 100% sure we want that behavior in this instance, since these are NIRISS observations that we may not want to have concatenated.

Reply:
Contents of the ZIP file (minus that random notebook that we don't need):

```python
['.\\canucs_basic_F150W_direct_dit1_i2d.fits',
 '.\\canucs_basic_F150W_WFSSC_dit1_cal.fits',
 '.\\canucs_basic_F150W_WFSSC_dit1_x1d.fits',
 '.\\canucs_basic_F150W_WFSSR_dit1_cal.fits',
 '.\\canucs_basic_F150W_WFSSR_dit1_x1d.fits']
```

For x1d , here is the FITS info:

```
Filename: .\canucs_basic_F150W_WFSSC_dit1_x1d.fits
No.    Name      Ver    Type      Cards   Dimensions   Format
  0  PRIMARY       1 PrimaryHDU     259   ()
  1  EXTRACT1D     1 BinTableHDU     47   107R x 9C   [D, D, D, D, D, J, D, D, D]
  2  EXTRACT1D     2 BinTableHDU     47   42R x 9C   [D, D, D, D, D, J, D, D, D]
  3  EXTRACT1D     3 BinTableHDU     47   110R x 9C   [D, D, D, D, D, J, D, D, D]
  4  EXTRACT1D     4 BinTableHDU     47   103R x 9C   [D, D, D, D, D, J, D, D, D]
  5  EXTRACT1D     5 BinTableHDU     47   107R x 9C   [D, D, D, D, D, J, D, D, D]
  6  EXTRACT1D     6 BinTableHDU     47   108R x 9C   [D, D, D, D, D, J, D, D, D]
  7  EXTRACT1D     7 BinTableHDU     47   103R x 9C   [D, D, D, D, D, J, D, D, D]
  8  EXTRACT1D     8 BinTableHDU     47   107R x 9C   [D, D, D, D, D, J, D, D, D]
  9  EXTRACT1D     9 BinTableHDU     47   93R x 9C   [D, D, D, D, D, J, D, D, D]
 10  EXTRACT1D    10 BinTableHDU     47   105R x 9C   [D, D, D, D, D, J, D, D, D]
 11  EXTRACT1D    11 BinTableHDU     47   108R x 9C   [D, D, D, D, D, J, D, D, D]
 12  EXTRACT1D    12 BinTableHDU     47   111R x 9C   [D, D, D, D, D, J, D, D, D]
 13  EXTRACT1D    13 BinTableHDU     47   110R x 9C   [D, D, D, D, D, J, D, D, D]
 14  EXTRACT1D    14 BinTableHDU     47   104R x 9C   [D, D, D, D, D, J, D, D, D]
 15  EXTRACT1D    15 BinTableHDU     47   111R x 9C   [D, D, D, D, D, J, D, D, D]
 16  EXTRACT1D    16 BinTableHDU     47   106R x 9C   [D, D, D, D, D, J, D, D, D]
 17  EXTRACT1D    17 BinTableHDU     47   101R x 9C   [D, D, D, D, D, J, D, D, D]
 18  EXTRACT1D    18 BinTableHDU     47   58R x 9C   [D, D, D, D, D, J, D, D, D]
 19  EXTRACT1D    19 BinTableHDU     47   111R x 9C   [D, D, D, D, D, J, D, D, D]
 20  EXTRACT1D    20 BinTableHDU     47   108R x 9C   [D, D, D, D, D, J, D, D, D]
 21  ASDF          1 BinTableHDU     11   1R x 1C   [107650B]
```

```
Filename: .\canucs_basic_F150W_WFSSR_dit1_x1d.fits
No.    Name      Ver    Type      Cards   Dimensions   Format
  0  PRIMARY       1 PrimaryHDU     259   ()
  1  EXTRACT1D     1 BinTableHDU     47   3R x 9C   [D, D, D, D, D, J, D, D, D]
  2  EXTRACT1D     2 BinTableHDU     47   19R x 9C   [D, D, D, D, D, J, D, D, D]
  3  EXTRACT1D     3 BinTableHDU     47   78R x 9C   [D, D, D, D, D, J, D, D, D]
  4  EXTRACT1D     4 BinTableHDU     47   107R x 9C   [D, D, D, D, D, J, D, D, D]
  5  EXTRACT1D     5 BinTableHDU     47   110R x 9C   [D, D, D, D, D, J, D, D, D]
  6  EXTRACT1D     6 BinTableHDU     47   107R x 9C   [D, D, D, D, D, J, D, D, D]
  7  EXTRACT1D     7 BinTableHDU     47   105R x 9C   [D, D, D, D, D, J, D, D, D]
  8  EXTRACT1D     8 BinTableHDU     47   112R x 9C   [D, D, D, D, D, J, D, D, D]
  9  EXTRACT1D     9 BinTableHDU     47   107R x 9C   [D, D, D, D, D, J, D, D, D]
 10  EXTRACT1D    10 BinTableHDU     47   109R x 9C   [D, D, D, D, D, J, D, D, D]
 11  EXTRACT1D    11 BinTableHDU     47   109R x 9C   [D, D, D, D, D, J, D, D, D]
 12  EXTRACT1D    12 BinTableHDU     47   116R x 9C   [D, D, D, D, D, J, D, D, D]
 13  EXTRACT1D    13 BinTableHDU     47   103R x 9C   [D, D, D, D, D, J, D, D, D]
 14  EXTRACT1D    14 BinTableHDU     47   111R x 9C   [D, D, D, D, D, J, D, D, D]
 15  EXTRACT1D    15 BinTableHDU     47   104R x 9C   [D, D, D, D, D, J, D, D, D]
 16  EXTRACT1D    16 BinTableHDU     47   101R x 9C   [D, D, D, D, D, J, D, D, D]
 17  EXTRACT1D    17 BinTableHDU     47   109R x 9C   [D, D, D, D, D, J, D, D, D]
 18  EXTRACT1D    18 BinTableHDU     47   107R x 9C   [D, D, D, D, D, J, D, D, D]
 19  EXTRACT1D    19 BinTableHDU     47   108R x 9C   [D, D, D, D, D, J, D, D, D]
 20  ASDF          1 BinTableHDU     11   1R x 1C   [101454B]
```

So 20 + 19 = 39. And I guess the other 1 or 2 links go to `i2d` and `cal` linking? When you load this dataset into Jdaviz manually, does it behave correctly? I assume each extension is a different slit, so when you go through the table, do things change accordingly?

Reply:
ok, then since this is an expected change by the change to specutils, #1752 updates the test to 40 (so that other PRs can rebase on that in the meantime until this change is merged) and you'll then just have to keep this change to 41 after rebasing on top of that.

Reply:
Oh wait, I thought this was Mosviz. But now I see it is Specviz. How did 39 `EXTRACT1D` spectra create 40-41 links? Shouldn't it be 38? 🤯

Reply:
Here are the data that it loads/concatenates: 
```
DataCollection (41 data sets)
	  0: NIRISS_for_parser_p0171 0
	  1: NIRISS_for_parser_p0171 1
	  2: NIRISS_for_parser_p0171 2
	  3: NIRISS_for_parser_p0171 3
	  4: NIRISS_for_parser_p0171 4
	  5: NIRISS_for_parser_p0171 5
	  6: NIRISS_for_parser_p0171 6
	  7: NIRISS_for_parser_p0171 7
	  8: NIRISS_for_parser_p0171 8
	  9: NIRISS_for_parser_p0171 9
	 10: NIRISS_for_parser_p0171 10
	 11: NIRISS_for_parser_p0171 11
	 12: NIRISS_for_parser_p0171 12
	 13: NIRISS_for_parser_p0171 13
	 14: NIRISS_for_parser_p0171 14
	 15: NIRISS_for_parser_p0171 15
	 16: NIRISS_for_parser_p0171 16
	 17: NIRISS_for_parser_p0171 17
	 18: NIRISS_for_parser_p0171 18
	 19: NIRISS_for_parser_p0171 19
	 20: NIRISS_for_parser_p0171 20
	 21: NIRISS_for_parser_p0171 21
	 22: NIRISS_for_parser_p0171 22
	 23: NIRISS_for_parser_p0171 23
	 24: NIRISS_for_parser_p0171 24
	 25: NIRISS_for_parser_p0171 25
	 26: NIRISS_for_parser_p0171 26
	 27: NIRISS_for_parser_p0171 27
	 28: NIRISS_for_parser_p0171 28
	 29: NIRISS_for_parser_p0171 29
	 30: NIRISS_for_parser_p0171 30
	 31: NIRISS_for_parser_p0171 31
	 32: NIRISS_for_parser_p0171 32
	 33: NIRISS_for_parser_p0171 33
	 34: NIRISS_for_parser_p0171 34
	 35: NIRISS_for_parser_p0171 35
	 36: NIRISS_for_parser_p0171 36
	 37: NIRISS_for_parser_p0171 37
	 38: NIRISS_for_parser_p0171 38
	 39: Combined [canucs_basic_F150W_WFSSR_dit1_x1d.fits]
	 40: Combined [canucs_basic_F150W_WFSSC_dit1_x1d.fits]
```